app/services/email_confirmation_token_validator.rb

```diff
@@ -7,7 +7,8 @@ class EmailConfirmationTokenValidator
   validate :email_not_already_confirmed, if: :email_address_found_with_token?
   validate :token_not_expired, if: :email_address_found_with_token?
 
-  def initialize(email_address)
+  def initialize(email_address, current_user = nil)
+    return unless current_user.nil? || email_address&.user == current_user
```

Comment:
early `return` in an initializer is really confusing to me, WDYT of saving `current_user` as an instance variable and adding a validation check?

Reply:
sounds good